test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Services/NuGetPackageFileServiceTests.cs

```diff
@@ -112,6 +111,47 @@ public async Task GetPackageIconAsync_EmbeddedFromFallbackFolder_HasOnlyReadAcce
             }
         }
 
+        [Fact]
+        public async Task GetPackageIconAsync_RemoteUriInaccessible_FindsLocalIconFromCache()
+        {
+            // Arrange
+            using (TestDirectory testDirectory = TestDirectory.Create())
+            {
+                string fileContents = "testFileContents";
+                string filePath = Path.Combine(testDirectory.Path, "testFile.txt");
+
+                File.WriteAllText(filePath, fileContents);
+
+                var packageFileService = new NuGetPackageFileService(
+                        default(ServiceActivationOptions),
+                        Mock.Of<IServiceBroker>(),
+                        new AuthorizationServiceClient(Mock.Of<IAuthorizationService>()),
+                        _telemetryProvider.Object);
+
+                var packageIdentity = new PackageIdentity("packageA", NuGetVersion.Parse("1.0.0"));
+                var remoteInaccessibleUri = new Uri("http://api.source/index.json");
+
+                // Add a fragment to consider this an embedded icon.
+                // Note: file:// is required for Uri to parse the Fragment (possibly a Uri bug).
+                var localUri = new Uri("file://" + filePath + "#testFile.txt");
+
+                NuGetPackageFileService.AddIconToCache(packageIdentity, remoteInaccessibleUri);
+
+                // Act
+                Stream inaccessibleRemoteIconStream = await packageFileService.GetPackageIconAsync(packageIdentity, CancellationToken.None);
+                NuGetPackageFileService.AddLocalIconToCache(packageIdentity, localUri);
+                Stream foundLocalIconStream = await packageFileService.GetPackageIconAsync(packageIdentity, CancellationToken.None);
```

Comment:
Since the file being opened exists inside the `TestDirectory` created at the start of this test, but these streams are not disposed, I expect that when the `TestDirectory` is disposed due to the `using` statement, that the directory delete fails because the file is in use. The test doesn't fail, because [the code used to delete the directories](https://github.com/NuGet/NuGet.Client/blob/8285b74c46ce9f01b1111c181bac535bc26aa9e0/test/TestUtilities/Test.Utility/TestFileSystemUtility.cs#L160-L180) silently ignores exceptions.

But the end result is that the test does not clean up after itself, which arguably is a test bug.

It's also a violation of the general coding convention that disposable objects must be disposed explicitly (and the `NuGetPackageFileService` is also not being disposed). While it might be questionable which coding standard/guidelines/recommendations we want to enforce in test code vs product code, the test directories not getting cleaned up is an unwanted side effect.

Reply:
Yes - I had noticed a cluttered test folder, but forgot to go back and cleanup the Streams. done now